src/customizations/volto/components/manage/Contents/ContentsUploadModal.jsx

```diff
@@ -0,0 +1,373 @@
+  }
+
+  /**
+   * Component will receive props
+   * @method componentWillReceiveProps
+   * @param {Object} nextProps Next properties
+   * @returns {undefined}
+   */
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    if (this.props.request.loading && nextProps.request.loaded) {
+      this.props.onOk();
+      this.setState({
+        files: [],
+      });
+    }
+  }
+
+  /**
+   * Remove file handler
+   * @method onRemoveFile
+   * @param {Object} event Event object
+   * @returns {undefined}
+   */
+  onRemoveFile(event) {
+    this.setState({
+      files: filter(
+        this.state.files,
+        (file, index) =>
+          index !== parseInt(event.target.getAttribute('value'), 10),
+      ),
+    });
+  }
+
+  /**
+   * Drop handler
+   * @method onDrop
+   * @param {array} files File objects
+   * @returns {undefined}
+   */
+  onDrop = async (files) => {
+    const validFiles = [];
+    for (let i = 0; i < files.length; i++) {
+      if (validateFileUploadSize(files[i], this.props.intl.formatMessage)) {
+        await readAsDataURL(files[i]).then((data) => {
+          const fields = data.match(/^data:(.*);(.*),(.*)$/);
+          files[i].preview = fields[0];
+        });
+        validFiles.push(files[i]);
+      }
+    }
+    this.setState({
+      files: concat(this.state.files, validFiles),
+    });
+  };
+
+  /**
+   * Cancel handler
+   * @method onCancel
+   * @returns {undefined}
+   */
+  onCancel() {
+    this.props.onCancel();
+    this.setState({
+      files: [],
+    });
+  }
+
+  /**
+   * Submit handler
+   * @method onSubmit
+   * @returns {undefined}
+   */
+  onSubmit() {
+    Promise.all(map(this.state.files, (file) => readAsDataURL(file))).then(
+      (files) => {
+        this.props.createContent(
+          this.props.pathname,
+          map(this.state.files, (file, index) => {
+            const fields = files[index].match(/^data:(.*);(.*),(.*)$/);
+            const image = fields[1].split('/')[0] === 'image';
+            return {
+              '@type': image ? 'Image' : 'File',
+              title: file.name,
+              [image ? 'image' : 'file']: {
+                data: fields[3],
+                encoding: fields[2],
+                'content-type': fields[1],
+                filename: file.name,
+              },
+            };
+          }),
+          SUBREQUEST,
+        );
+      },
+    );
+  }
+
+  componentDidMount() {
+    this.props.getTypes(getBaseUrl(this.props.pathname));
+  }
+
+  /**
+   * Render method.
+   * @method render
+   * @returns {string} Markup for the component.
+   */
+  render() {
+    // as per agid guidelines, files cannot be uploaded in modulistica folder inside ct servizio
+    // show restraint and hide upload button when page is called modulistica and when there's a restraint
+    // this is enough to identify this only case bc even if another page is called 'modulistica', it will have 'File'
+    // as addable type
+    const showFileRestraint =
```

Comment:
Se l'obiettivo e' verificare che non sia negli allowed, ti propongo l'inverso con some, cosi' non giri inutilmente tutto l'array, se lo becchi esci subito
`!this.props.types.some(type => type.id === "File")`
Per il 98% dei nostri casi, esci sempre prima